src/pkgcore/ebuild/eclass.py

```diff
@@ -235,6 +235,29 @@ def _supported_eapis(self, block, tag, lineno):
 
 
 class EclassVarBlock(ParseEclassDoc):
+    """ECLASS_VARIABLE doc block."""
+
+    tag = '@ECLASS_VARIABLE:'
+    key = 'variables'
+    default = True
+
+    def __init__(self):
+        tags = {
+            '@ECLASS_VARIABLE:': ('name', True, self._tag_inline_arg, None),
+            '@DEPRECATED:': ('deprecated', False, self._tag_deprecated, False),
+            '@DEFAULT_UNSET': ('default_unset', False, self._tag_bool, False),
+            '@INTERNAL': ('internal', False, self._tag_bool, False),
+            '@REQUIRED': ('required', False, self._tag_bool, False),
+            '@PRE_INHERIT': ('pre_inherit', False, self._tag_bool, False),
+            '@USER_VARIABLE': ('user_variable', False, self._tag_bool, False),
+            '@OUTPUT_VARIABLE': ('output_variable', False, self._tag_bool, False),
+            '@DESCRIPTION:': ('description', True, self._tag_multiline_str, None),
+        }
+        super().__init__(tags)
+
+
+# For backwards compatibility, can be removed after 2022-12-31
+class EclassVarBlockCompat(ParseEclassDoc):
```

Comment:
Hmm, no chance you could use `EclassVarBlock` as the base class and just override `tag`?

Reply:
That's why I had asked in #-qa. :smile:
```
<@mgorny> i think duplicating is fine
<@mgorny> or even better
```

`@ECLASS-VARIABLE` is also used in `tags`, so would have to override that too (how?).

Reply:
Sorry, but I fear that my python skills aren't sufficient to produce a more elegant solution. Especially, not sure how the child class would access `tags`, which is defined inside the `__init__` method of its parent.